2996_reflection/reflection.md

```diff
@@ -3477,49 +3547,67 @@ Add a new paragraph between [expr.eq]{.sref}/5 and /6:
 
 ### [expr.const]{.sref} Constant Expressions {-}
 
-Add new paragraphs prior to the definition of _manifestly constant evaluated_ ([expr.const]{.sref}/20), and renumber accordingly:
+Modify paragraph 15 to disallow returning non-consteval-only pointers and references to consteval-only objects from constant expressions.
 
 ::: std
-::: addu
+[15]{.pnum} A _constant-expression_ is either a glvalue core constant expression that [\ ]{.addu}
 
-[20]{.pnum} An expression or conversion is _in substitution context_ if it results from the substitution of template parameters
+* [#.#]{.pnum} refers to an entity that is a permitted result of a constant expression[, and]{.addu}
+* [[#.#]{.pnum} is of consteval-only type if the entity designated by the expression is of consteval-only type,]{.addu}
 
-* [#.#]{.pnum} during template argument deduction ([temp.deduct]{.sref}),
-* [#.#]{.pnum} in a `$concept-id$` ([temp.names]{.sref}), or
-* [#.#]{.pnum} in a `$requires-expression$` ([expr.prim.req]{.sref}).
+or a prvalue core constant expression whose value satisfies the following constraints:
+
+* [#.#]{.pnum} if the value is an object of class type, each non-static data member of reference type refers to an entity that is a permitted result of a constant expression [and has consteval-only type if the entity has consteval-only type]{.addu},
+* [#.#]{.pnum} if the value is an object of scalar type, it does not have an indeterminate or erroneous value ([basic.indet]),
+* [#.#]{.pnum} if the value is of pointer type, it is a pointer to an object of static storage duration, a pointer past the end of such an object ([expr.add]), a pointer to a non-immediate function, or a null pointer value,
+* [[#.#]{.pnum} if the value is a pointer to an object of consteval-only type, or a pointer past the end of such an object, then the prvalue is also of consteval-only type,]{.addu}
```

Comment:
this should be part of the previous bullet somehow. Maybe like:

> * if the value is of pointer type, then:
>   * it is a pointer to, or past the end of, object of static storage duration and, if the object has consteval-only type, the value also has consteval-only type, or
>   * it is a pointer to a non-immediate function, or
>   * it is a null pointer value,

Ish?

Reply:
Done.